File: java/src/main/java/com/iteso/nintendo/WaterGun.java
package com.iteso.nintendo;
/**Ataque Water Gun.*/
public class WaterGun implements Attack {

    /**Nombre del ataque.*/
    private final String name = "Ataque Water Gun";
    /**Daño del ataque.*/
    private final int damage = 25;

    /**@return Nombre del ataque.*/
    @Override
    public String getAttackName() {
        return this.name;
    }

    /**@return Daño del ataque.*/
    @Override
    public int getAttackDamage() {
        return this.damage;
    }
}

File: csharp/Clases/WaterGun.cs
namespace unittestpractice.Clases
{
    public class WaterGun : IAttack
    {

        /**Nombre del ataque.*/
        private readonly String _name = "Ataque Water Gun";
        /**Daño del ataque.*/
        private const int Damage = 25;

        public String GetAttackName()
        {
            return _name;
        }

        public int GetAttackDamage()
        {
            return Damage;
        }
    }
}